frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -1,41 +1,54 @@
 package bloop.engine
 
 import bloop.cli.{CliOptions, Commands, CommonOptions, ExitStatus}
+import bloop.io.SourceWatcher
 import bloop.io.Timer.timed
 import bloop.reporter.ReporterConfig
 import bloop.engine.tasks.{CompileTasks, TestTasks}
 import bloop.Project
+import bloop.logging.BloopLogger
 
 object Interpreter {
-  def execute(action: Action, state: State): State = {
-    import state.logger
+  def execute(action: Action, state0: State): State = {
+    import state0.logger
     def logAndTime[T](cliOptions: CliOptions, action: => T): T =
-      logger.verboseIf(cliOptions.verbose) { timed(state.logger) { action } }
+      logger.verboseIf(cliOptions.verbose) { timed(state0.logger) { action } }
+    def updateState(state: State, commonOptions: CommonOptions): State = {
+      State.updateLogger(state.logger, commonOptions)
+      val logger = BloopLogger(state.logger.name)
```

Comment:
We need a new logger because log4j forces us to call the logger again to get the changes we just made, that involves calling LogManager.getLogger (which is done in the constructor of BloopLogger).

Reply:
You should think of the logger as immutable, and that's why it's important to import it always from the current state.

Reply:
We do it because we need to call `LogManager.getLogger` every time we modify it (because apparentely changes are not propagated by log4j).